be/src/storage/rowset/segment_v2/data_decoder.cpp

```diff
@@ -0,0 +1,59 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021 StarRocks Limited.
+
+#include "storage/rowset/segment_v2/data_decoder.h"
+
+namespace starrocks::segment_v2 {
+
+std::unique_ptr<DataDecoder> get_data_decoder(EncodingTypePB encoding) {
+    switch (encoding) {
+    case BIT_SHUFFLE: {
+        return std::make_unique<BitShuffleDecoder>();
+    }
+    case FOR_ENCODING: {
+        return std::make_unique<FrameOfReferenceDecoder>();
+    }
+    case PLAIN_ENCODING: {
+        return std::make_unique<BinaryPlainDecoder>();
+    }
+    case DICT_ENCODING: {
+        return std::make_unique<BinaryDictDecoder>();
+    }
+    case PREFIX_ENCODING: {
+        return std::make_unique<BinaryPrefixDecoder>();
+    }
+    case RLE: {
+        return std::make_unique<RleDecoder>();
+    }
+    case DEFAULT_ENCODING: {
+        return std::make_unique<DefaultEncodingDecoder>();
+    }
+    default: {
+        return nullptr;
+    }
+    }
+}
+
+Status DataDecoder::decode_page(PageFooterPB* footer, uint32_t footer_size, EncodingTypePB encoding,
+                                std::unique_ptr<char[]>* page, Slice& page_slice) {
+    CHECK(footer->has_type()) << "type must be set";
+    switch (footer->type()) {
+    case INDEX_PAGE:
+    case DICTIONARY_PAGE:
+    case SHORT_KEY_PAGE: {
+        return Status::OK();
+    }
+    case DATA_PAGE: {
+        std::unique_ptr<DataDecoder> decoder = get_data_decoder(encoding);
+        // Unknown encoding type will return success directly
+        if (!decoder) {
+            return Status::OK();
```

Comment:
Q: How can this work correctly?

Reply:
> 

what i think is data_decoder is only responsible for decoding the data, if encoding_type is unknown, return the original data directly.

Reply:
> 

what i think is data_decoder is only responsible for decoding data, if encoding_type is unknown, return the original data directly

Reply:
If callers can handle these correctly,  there will be no necessity for them to call this method at all.

Reply:
> 

fix in https://github.com/StarRocks/starrocks/pull/1597/commits/e6056a6901ae3441973adfb4d58682166c78e2af, PTAL